nucliadb/nucliadb/export_import/datamanager.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+from typing import AsyncGenerator, AsyncIterator, Callable
+
+from aiofiles.threadpool import AsyncFileIO  # type: ignore
+from nucliadb_protos.resources_pb2 import CloudFile
+from nucliadb_protos.writer_pb2 import (
+    BrokerMessage,
+    GetEntitiesResponse,
+    GetLabelsResponse,
+)
+
+from nucliadb.common.maindb.driver import Driver
+from nucliadb.ingest.orm.entities import EntitiesManager
+from nucliadb.ingest.orm.knowledgebox import KnowledgeBox as KnowledgeBoxORM
+from nucliadb_utils.const import Streams
+from nucliadb_utils.partition import PartitionUtility
+from nucliadb_utils.storages.storage import Storage
+from nucliadb_utils.transaction import TransactionUtility
+
+# Broker message fields that are populated by the processing pipeline
+PROCESSING_BM_FIELDS = [
+    "link_extracted_data",
+    "file_extracted_data",
+    "extracted_text",
+    "field_metadata",
+    "field_vectors",
+    "field_large_metadata",
+    "user_vectors",
+]
+
+# Broker message fields that are populated by the nucliadb writer component
+WRITER_BM_FIELDS = [
+    "links",
+    "files",
+    "texts",
+    "conversations",
+    "layouts",
+    "keywordsets",
+    "datetimes",
+]
+
+BinaryStream = AsyncGenerator[bytes, None]
+BinaryStreamGenerator = Callable[[int], BinaryStream]
+
+
+class KBExporterDataManager:
+    def __init__(self, driver: Driver, storage: Storage):
+        self.driver = driver
+        self.storage = storage
+
+    async def iter_broker_messages(self, kbid: str) -> AsyncIterator[BrokerMessage]:
+        async with self.driver.transaction() as txn:
+            kb = KnowledgeBoxORM(txn, self.storage, kbid)
+            async for resource in kb.iterate_resources():
+                resource.disable_vectors = False
+                bm = await resource.generate_broker_message()
+                yield bm
+
+    def get_binaries(self, bm: BrokerMessage) -> list[CloudFile]:
```

Comment:
I was thinking data managers should focus on isolating the interaction with the database. Maybe this function doesn't fit?